com.bambulab.BambuStudio.yml

```diff
@@ -0,0 +1,382 @@
+      - -DGLFW_USE_OSMESA=OFF
+    sources:
+      - type: git
+        url: https://github.com/glfw/glfw.git
+        commit: 3fa2360720eeba1964df3c0ecf4b5df8648a8e52
+      - type: patch
+        path: patches/0003-Don-t-crash-on-calls-to-focus-or-icon.patch
+      - type: patch
+        path: patches/0005-Add-warning-about-being-an-unofficial-patch.patch
+      - type: patch
+        path: patches/0007-Platform-Prefer-Wayland-over-X11.patch
+    cleanup:
+      - /include
+      - /lib/cmake
+      - /lib/pkgconfig
+
+  - name: BambuStudio
+    buildsystem: simple
+    build-commands:
+      # start build
+      - |
+        mkdir -p deps/build && cd deps/build
+        cmake ../ \
+          -DDEP_WX_GTK3=1 \
+          -DDEP_DOWNLOAD_DIR=/run/build/BambuStudio/external-packages
+        cmake --build . --target dep_Boost dep_TBB dep_wxWidgets dep_Cereal dep_NLopt dep_OpenVDB dep_Qhull dep_OpenCSG dep_CGAL dep_OCCT
+      - |
+        mkdir -p build && cd build
+        CXXFLAGS=-std=gnu++20 cmake ../ \
+          -GNinja \
+          -DCMAKE_INSTALL_PREFIX=/app \
+          -DCMAKE_INSTALL_LIBDIR=/app/lib \
+          -DCMAKE_PREFIX_PATH=/run/build/BambuStudio/deps/build/destdir/usr/local \
+          -DSLIC3R_PCH=OFF \
+          -DSLIC3R_FHS=ON \
+          -DSLIC3R_GTK=3 \
+          -DSLIC3R_STATIC=ON \
+          -DSLIC3R_BUILD_TESTS=OFF \
+          -DSLIC3R_DESKTOP_INTEGRATION=OFF \
+          -DCMAKE_BUILD_TYPE=Release
+        CXXFLAGS=-std=gnu++20 cmake --build . --target install
+
+    post-install:
+
+      - | # locale l10n
+        mkdir -p /app/share/runtime/locale
+        for i in $(ls /app/share/PrusaSlicer/localization)
+        do
+          lang=${i%[_@]*}
+          mkdir -p /app/share/runtime/locale/${lang}
+          mv /app/share/PrusaSlicer/localization/${i} /app/share/runtime/locale/${lang}
+          ln -rs /app/share/runtime/locale/${lang}/${i} /app/share/PrusaSlicer/localization/${i}
+        done
+
+      - | # Desktop Integration files
+        install -Dm644 com.bambulab.BambuStudio.metainfo.xml /app/share/metainfo/com.bambulab.BambuStudio.metainfo.xml
+        install -Dm644 BambuStudio_512x512x32.png /app/share/icons/hicolor/512x512/apps/com.bambulab.BambuStudio.png
+        install -Dm644 BambuStudio_256x256x32.png /app/share/icons/hicolor/256x256/apps/com.bambulab.BambuStudio.png
+        install -Dm644 src/platform/unix/BambuStudio.desktop /app/share/applications/com.bambulab.BambuStudio.desktop
+        install set-dark-theme-variant.py /app/bin
+        install entrypoint /app/bin
+        install umount /app/bin
+
+    sources:
+      # -
+      # Section bellow fetches all BambuStudio dependencies before the build process and stores them in external-packages/*/* .
+      # -DDEP_DOWNLOAD_DIR is set in the build process which has to match with dest.
+      #
+      # NOTE: The url, dest folder name and sha256 must match from BambuStudio's cmake scripts and folder names in BambuStudio/deps/
+      # -
```

Comment:
can't we avoid building those libraries in the runtime?

Reply:
A lot of those are vendored, and because the codebase relies on specific version, it's easier seeing a build failure from a missing cached archives than tracking changes down manually.

That being said, I've de-vendored, and either relied on the system deps or build separately:
- wxwidgets (we needed a newer version to have EGL support anyway)
- boost
- libjpeg
- libpng
- libtiff
- boost

I still need to make use of the system deps for:
- expat
- freetype
- openssl
- glew
- zlib

I don't intend to make any changes to the other deps.